src/Presentation/ListFacetHtmlBuilder.php

```diff
@@ -33,37 +38,63 @@ public function buildHtml( FacetConfig $config, PropertyConstraints $state ): st
 	 * @return array<string, mixed>
 	 */
 	public function buildViewModel( FacetConfig $config, PropertyConstraints $state ): array {
-		$combineWithAnd = true; // TODO: use state and config defaultCombineWith
+		$combineWithAnd = $this->shouldCombineWithAnd( $config, $state );
 
 		return [
-			'toggle' => $this->buildToggleViewModel( $config, $state, $combineWithAnd ),
-			'checkboxes' => $this->buildCheckboxesViewModel( $config, $state, $combineWithAnd ),
+			'toggle' => $this->buildToggleViewModel( $combineWithAnd, $this->hasCombineWithChoice( $config ) ),
+			'checkboxes' => $this->buildCheckboxesViewModel( $config, $state ),
 			// TODO: act on config: showNoneFilter
 			// TODO: act on config: showAnyFilter
 		];
 	}
 
+	public function shouldCombineWithAnd( FacetConfig $config, PropertyConstraints $state ): bool {
+		if ( $this->hasCombineWithChoice( $config ) ) {
+			if ( $state->getOrSelectedValues() !== [] ) {
+				return false;
+			}
+
+			if ( $state->getAndSelectedValues() !== [] ) {
+				return true;
+			}
+		}
+
+		if ( ( $config->typeSpecificConfig[self::CONFIG_KEY_DEFAULT_COMBINE_WITH] ?? null ) === 'OR' ) {
+			return false;
+		}
+
+		if ( ( $config->typeSpecificConfig[self::CONFIG_KEY_DEFAULT_COMBINE_WITH] ?? null ) === 'AND' ) {
```

Comment:
I originally did not have the brackets around the `foo ?? null`. Another case of "if it is not tested, it is broken" :)